scripts/generator/generator/pyclblast.py

```diff
@@ -0,0 +1,113 @@
+
+# This file is part of the CLBlast project. The project is licensed under Apache Version 2.0. This file follows the
+# PEP8 Python style guide and uses a max-width of 120 characters per line.
+#
+# Author(s):
+#   Cedric Nugteren <www.cedricnugteren.nl>
+
+NL = "\n"
+SEPARATOR = "####################################################################################################"
+
+
+def to_np_dtype(flavour):
+    if flavour.precision_name == "S":
+        return "float32"
+    if flavour.precision_name == "D":
+        return "float64"
+    if flavour.precision_name == "C":
+        return "complex64"
+    if flavour.precision_name == "Z":
+        return "complex128"
+    raise RuntimeError("Could not convert flavour '%s' to numpy" % flavour.precision_name)
+
+
+def scalar_cython_conversion(scalar, flavour):
+    scalar_type = flavour.alpha_cl if scalar == "alpha" else flavour.beta_cl
+    if scalar_type == "float":
+        return "<cl_float>" + scalar
+    if scalar_type == "double":
+        return "<cl_double>" + scalar
+    if scalar_type in ["cl_float2", "float2"]:
```

Comment:
This one might be harder to put in a dict like above. If you wanted you could compare on the ending with 'value'.endswidth and construct the dict each time you enter the function.